secretupdater/secretupdater/models.py

```diff
@@ -288,6 +290,48 @@ def _get_k8s_auth(credentials):
                     }
                 ],
             }
+        elif _get_credential(kube, 'eks-certificate-authority-data'):
+            app.logger.debug("%s: Found EKS CA data" % (id))
+            eks_cluster_id = kube.get('metadata').get('eks-cluster-id')
+            eks_assume_role_arn = kube.get('metadata').get('eks-assume-role-arn') 
+            eks_output = sp.getoutput("aws eks get-token --cluster-name" + " " + eks_cluster_id + " " +  "--role" + " " + eks_assume_role_arn)
+            app.logger.debug(eks_output)
+            try: 
+                json_token = json.loads(eks_output) 
+                token = list(json_token.values())[3]['token']
+                app.logger.debug(token)
+            except ValueError:
+                raise BadEKSToken("Unable to parse token from aws cli")
+
+            config = {
+                "clusters": [
+                    {
+                        "name": id,
+                        "cluster": {
+                            "server": kube.get('metadata').get('endpoint'),
```

Comment:
maybe move the `.get`(s) up to line ~295 inside the new `try` block to capture configuration errors?